src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,215 @@
+        }
+        insertDrivers(car);
+        deleteDriversFromCar(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = FALSE "
+                + "WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t delete car in database", e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Car> carList = new ArrayList<>();
+        String query = "SELECT model, manufacturer_id FROM cars c JOIN cars_drivers cd "
+                + "ON c.id = cd.car_id WHERE cd.driver_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, driverId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                carList.add(getCar(resultSet));
+            }
+            for (Car car : carList) {
+                car.setDrivers(getDriversOfCar(car.getId()));
+            }
+            return carList;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get car from database with driver id: "
+                    + driverId, e);
+        }
+    }
+
+    private List<Driver> getDriversOfCar(Long carId) {
+        List<Driver> driverList = new ArrayList<>();
+        String query = "SELECT * FROM drivers d JOIN cars_drivers cd "
+                + "ON d.id = cd.driver_id WHERE cd.car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, carId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                driverList.add(getDrivers(resultSet));
+            }
+            return driverList;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get drivers from database with car id: "
+                    + carId, e);
+        }
+    }
+
+    private void insertDrivers(Car car) {
+        String query = "INSERT INTO cars_drivers (car_id, driver_id) "
+                + "VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, car.getId());
+            if (car.getDrivers() != null) {
```

Comment:
let's move this check, and also add a size check so that we won't even open a connection if it is null or empty